modules/ROOT/pages/continuous-integration.adoc

```diff
@@ -0,0 +1,67 @@
+= Continuous Integration
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: build, deploy, test, maven, ci, continuous integration, continuous, integration
+
+The need for continuous integration (CI) for a project is very important. By using Maven as your build tool, you can create a build that gets triggered on every project change, and run all of its unit and functional tests automatically.
+
+The advantages of CI are:
+
+* Provides early notification of issues in the software development lifecycle.
+* Ensures code gets fully tested before release.
+* Successfully tested branches ensure better success when merging to the master branch.
+
+== Continuous Integration Objectives
+
+A CI system does all of the following:
+
+. Listens for new commits to a project's source code management system. The CI system watches many branches for new commits. You can either use polling to find new commits, or the management system can trigger events that inform your program of commits.
+. Pulls the newest branch into a centralized server.
+. Creates build jobs on a centralized server.
+. Runs configurable unit and integration tests on the code base that compile, test, package, and deploy the project in a sandbox to ensure the project works correctly.
+. Stores artifacts in a repository.
+. Displays the results of each build.
+. Deploys passing builds to production.
+
+For information about these tools, see their websites.
+
+== MuleSoft CI Tools
+
+You can deploy Mule applications using:
+
+* xref:api-manager::getting-started-proxy.adoc[The API Manager]
+* xref:runtime-manager::runtime-manager-agent.adoc[The Runtime Manager Agent]
+
+You can create functional tests with xref:2.1@munit::index.adoc[MUnit Unit Testing].
+
+The mule-maven-plugin supports deployments to:
+
+* CloudHub
+* Anypoint Runtime Manager
+* Mule Agent
+* Local instances
+
+For a complete CI example that uses Jenkins and Maven, see the Anypoint Platform Development: Advanced class on the https://training.mulesoft.com/course-catalog[MuleSoft training site]. The training site also includes related classes and certification exams.
+
+== External CI Tools
+
+Popular CI tools are:
+
+* Jenkins
+* Hudson
+* TeamCity
+* Bamboo
+
+== Integration Notes
+
+If you've chosen to use real transports in your functional test cases, watch out for potential port conflicts that can occur in a continuous build server.
+
+If your target deployable is a web application and not a Mule application, consider using Junit Integration Test Runner for running your functional tests and avoiding port conflicts.
+
+NOTE: If you create Maven projects from the command line using archetypes, you must manage your POM file manually, and in some cases, adjust your `settings.xml` file to point to the MuleSoft Enterprise repository and supply credentials. In some cases, even if you manage your project with Anypoint Studio, you might need to make manual adjustments to your POM file as well.
```

Comment:
I'd remove this note because creating Maven projects from the command line using archetypes is not supported in Mule 4.